tutorials/10_DebuggingREMIND.md

```diff
@@ -134,8 +137,29 @@ If infeasibilities show up already in the first iteration, it may be related to
 
 There are different types of solver infeasibilities: pre-triangular and optimization infeasibilities. In pre-triangular infeasibilities, GAMS shows you in the solution report the equations that are incompatible with each other. For optimization infeasibilies, the CONOPT solver tries to reduce the infeasibility to the thing that less affects the objective function. It does not show all affected equations, as it is not a simple problem as a non-square system of equations like in the pre-triangular case. You need to check if it is bound-related: the variables bounds, and the equation bounds starting from the infeasibility and going through the variables that have relation with it. You can always force in another run the variable that is infeasible to a feasible value to see what else is affected by it. But this is usually not necessary as just checking the logic behind the equation and the infeasible variable is usually sufficient to find the limitation.
 
+Case 4: REMIND did not converge
+-------------------------------
+
+If the iterations are not sufficient to converge, you will see in the output of `rs2 -c` that `RunStatus` is `Normal completion`, but `Conv` shows `not_converged`.
+
+You can run
+```
+less -j 7 +?diagnostics full.lst
+```
+in your output subfolder to see the "Convergence diagnostics" of the last iteration (`+?diagnostics` searches for the last occurrence of `diagnostics`). Below, you should find reasons for non-convergence.
+
+If trade is the issue, search with `?p80_messageFailedMarket` to see how the failed markets evolved over the iterations. Looking at `p80_defic_sum_rel` will show you how the sum over all market imbalances developped over the iterations. Often, it improves a lot until iteration 20-30, then either converges or becomes worse again for 10-30 iterations, before improving again for 10-20 iterations, then becoming worse again. One can look which iteration in the range 20..35 was best, then go back to that iteration by searching backward for `o_iterationNumber` to check which markets were above the threshold then, comparing the values plotted below `p80_surplusMax2100` with `p80_surplusMaxTolerance`.
+
+If the run was mostly converged in that iteration (which depends on your standards), one can set `cm_nash_autoconverge` to `3`, which relaxes the requirements on market clearing further. This should then lead to the restarted run actually converging in that iteration. You can run REMIND for more iterations by modifying `cm_iteration_max` in [`80_optimization/nash/datainput.gms`](../modules/80_optimization/nash/datainput.gms) and the set iteration in [`core/sets.gms`](../core/sets.gms), which by default only goes to 200.
```

Comment:
this is not recommended for scenarios for paper or project submission. Maybe also the existing solution is sufficient for the purpose if only a few markets fail in late time steps. Would be even faster than restarting with lower tolerance criteria

Reply:
I would propose to add "This should not be used for paper runs, but can be ok for project runs if p80_defic_sum_rel is low (<0.02 or so)".

Often enough, the solution in iteration 100 can be quite bad, which is why I would usually rather go for restarting with relaxed limits. On using the run for a project: To my understanding the current thresholds are arbitrary, so doubling increases the imbalance, but doesn't necessarily make the runs "wrong" or "bad". I would say for almost all project uses this is totally sufficient - only when the paper submission time approaches it might be safer to go for a tighter convergence criterion.